service/k8sclient/kubectl/kubectl.go

```
@@ -40,24 +48,106 @@ type KubeCtl struct {
 
 // NewKubeCtl creates a new KubeCtl object with a given logger.
 func NewKubeCtl(l logger.Logger) *KubeCtl {
-	// TODO Handle kubectl versions https://jira.percona.com/browse/PMM-6348
-
-	// TODO Handle kubeconfig https://jira.percona.com/browse/PMM-6347
-
-	cmd := []string{"dbaas-kubectl-1.16"}
-	path, err := exec.LookPath(cmd[0])
-	l.Debugf("path: %s, err: %v", path, err)
-	if e, ok := err.(*exec.Error); err != nil && ok && e.Err == exec.ErrNotFound {
-		cmd = []string{"minikube", "kubectl", "--"}
+	// Handle kubectl versions
+	cmd, err := getKubectlCmd()
+	if err != nil {
+		l.Debugf("Cannot find kubectl binary: %s", err)
+		return nil
```

Let's return that error and handle it in the caller